### PR TITLE
feature/10 - Build Native Images

### LINKED ISSUE #10: Build a native image
BFSC is currently delivered as an executable JAR that requires installing a JVM for running. We should build and deliver a native executable for macOS, Linux and Windows so that users don't need to install a JVM for running BFSC.

## Acceptance Criteria
1. An executable for macOS, Linux and Windows is uploaded to the release page (in addition to the jar)
2. Integration tests verify that the executable works on all three platforms
3. User guide describes how to use the executable

See https://github.com/exasol/parquet-edml-generator/pull/20 for an example

### PR DESCRIPTION
**Pull request by ckunki**
fixes #10

### CODE REVIEW

File: doc/changes/changes_1.1.0.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Bucketfs Client 1.1.0, released 2023-??-??
+
+Code name: Added native image
+
+## Summary
+
+Enhanced build scripts to build native images of BFSC for various operating systems.
```
Comment (kaklakariada) [inline doc/changes/changes_1.1.0.md:7]:
```suggestion
Code name: Added native executables

## Summary

Enhanced build scripts to build native executables of BFSC for operating systems Linux, macOS and Windows.
```

File: src/test/java/com/exasol/bucketfs/native_image/NativeImageTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.junit.jupiter.api.Test;
+
+import com.exasol.bucketfs.JarExecutor;
+
+@Tag("native-image")
+class NativeImageTest {
+
+    @Test
+    void testNativeImage() throws IOException, InterruptedException {
+        final JarExecutor executor = new JarExecutor().run("--help");
```
Comment (kaklakariada) [inline src/test/java/com/exasol/bucketfs/native_image/NativeImageTest.java:18]: This executes the JAR, not the native image!

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/bucketfs/native_image/NativeImageTest.java (hunk 2 above)
Comment (ckunki) [reply src/test/java/com/exasol/bucketfs/native_image/NativeImageTest.java:22]: Oh thanks - I need to fix this.

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 3]
```diff
@@ -14,45 +14,67 @@ jobs:
...
+      - uses: graalvm/setup-graalvm@v1
         with:
-          distribution: 'temurin'
-          java-version: 11
-          cache: 'maven'
+          version: '22.3.1'
+          java-version: '11'
+          components: 'native-image'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          native-image-job-reports: 'true'
```
Comment (kaklakariada) [inline .github/workflows/release_droid_upload_github_release_assets.yml:36]: 👍

File: doc/developers_guide/building_native_executables.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# Building Native Executables
```
Comment (kaklakariada) [inline doc/developers_guide/building_native_executables.md:1]: ❤️

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**